pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -689,17 +782,21 @@ func needsRetry(shoot *gardencorev1beta1.Shoot) bool {
 	return needsRetryOperation
 }
 
-func getOperation(shoot *gardencorev1beta1.Shoot) string {
-	var (
-		operation            = v1beta1constants.GardenerOperationReconcile
-		maintenanceOperation = shoot.Annotations[v1beta1constants.GardenerMaintenanceOperation]
-	)
+func getOperation(shoot *gardencorev1beta1.Shoot, credentialsRotationUpdate map[string]updateResult) string {
+	maintenanceOperation := shoot.Annotations[v1beta1constants.GardenerMaintenanceOperation]
+
+	for credentials := range credentialsRotationUpdate {
+		if (credentials == sshKeypair && maintenanceOperation == v1beta1constants.ShootOperationRotateSSHKeypair) ||
+			(credentials == observabilityPasswords && maintenanceOperation == v1beta1constants.OperationRotateObservabilityCredentials) {
+			return v1beta1constants.GardenerOperationReconcile
```

Comment:
Could you add a comment why this is necessary? E.g. I guess it is because we already patch the status indicating that the credential rotation was started and don't want to patch it again when the reconciliations tarts.